npc_simulator/src/node.cpp

```diff
@@ -897,55 +931,55 @@ void NPCSimulatorNode::objectCallback(const npc_simulator::Object::ConstPtr & ms
       tf2::Transform tf_input2map;
 
       try {
-        geometry_msgs::TransformStamped ros_input2map =
-          tf_buffer_.lookupTransform(
-            msg->header.frame_id, "map", msg->header.stamp, ros::Duration(0.5));
+        geometry_msgs::msg::TransformStamped ros_input2map = tf_buffer_.lookupTransform(
+          msg->header.frame_id, "map", msg->header.stamp, rclcpp::Duration(0.5));
         tf2::fromMsg(ros_input2map.transform, tf_input2map);
       } catch (tf2::TransformException & ex) {
-        ROS_WARN("%s", ex.what());
+        RCLCPP_WARN(get_logger(), "%s", ex.what());
         return;
       }
 
       tf2::Transform tf_input2object_origin;
       tf2::fromMsg(msg->initial_state.pose_covariance.pose, tf_input2object_origin);
 
-      npc_simulator::Object object = *msg;
+      npc_simulator::msg::Object object = *msg;
       object.header.frame_id = "map";
 
       tf2::Transform tf_map2object_origin = tf_input2map.inverse() * tf_input2object_origin;
       tf2::toMsg(tf_map2object_origin, object.initial_state.pose_covariance.pose);
       // publish
-      dummy_perception_object_pub_.publish(convertObjectMsgToDummyPerception(&object));
+      dummy_perception_object_pub_->publish(convertObjectMsgToDummyPerception(&object));
       objects_.push_back(object);
       break;
     }
-    case npc_simulator::Object::DELETE: {
+    case npc_simulator::msg::Object::DELETE: {
       for (size_t i = 0; i < objects_.size(); ++i) {
         if (objects_.at(i).id.uuid == msg->id.uuid) {
           // publish
-          objects_.at(i).action == npc_simulator::Object::DELETE;
```

Comment:
Bug before was `==` instead of `=`. Caught by -Werror=unused-value

Reply:
Oof!